kasa/smart/smartdevice.py

```diff
@@ -126,8 +128,10 @@ async def update(self, update_children: bool = True):
             if not self.children:
```

Comment:
Is there any use case where someone would want to call update and not update the children?  Seems harmless checking it anyway and it's less confusing than a parameter that doesn't do anything

Reply:
So this is currently used by homeassistant to avoid updating the child devices on HS300. Basically, it uses the regular updates for the main device, but has separate update coordinators with a higher update interval to fetch the (emeter) data for each of the socket.

I'd love to get rid of that special handling to simplify the homeassistant integration, but I'm unsure if there's a nice way for that :-(